src/core/buffer/representation_buffer.ts

```diff
@@ -294,20 +308,59 @@ export default function RepresentationBuffer<T>({
           currentSegmentRequest = null;
           return Observable.empty();
         }
+
+        const { initSegment } = initSegmentObject;
         const { segment, priority } = currentNeededSegment;
-        const initSegmentInfos = initSegmentObject && initSegmentObject.segmentInfos;
         const request$ = segmentFetcher.createRequest({
           adaptation,
-          init: initSegmentInfos || undefined,
+          init: initSegmentObject.value.segmentInfos || undefined,
           manifest,
           period,
           representation,
           segment,
         }, priority);
 
         currentSegmentRequest = { segment, priority, request$ };
-        return request$
-          .map((args) => objectAssign({ segment }, args))
+
+        if (initSegmentObject.isDownloaded || initSegment == null) {
+          return request$
+            .map((args) => objectAssign({ segment }, args))
+            .concat(requestNextSegment$);
+        }
+
+        const initSegmentRequest$ = segmentFetcher.createRequest({
+          adaptation,
+          init: undefined,
+          manifest,
+          period,
+          representation,
+          segment: initSegment,
+        }, 0);
+
+        return Observable.merge(
+          initSegmentRequest$
+            .map((args) => objectAssign({ segment: initSegment }, args)),
+          request$
+            .map((args) => objectAssign({ segment }, args))
+        )
+          // Emit all segments once the init segment has finished to download
+          // and always emit the init segment first.
+          .scan((segments, obj) => {
+            if (initSegmentObject.isDownloaded) {
+              return [obj];
+            }
+            if (obj.segment.isInit) {
+              initSegmentObject.isDownloaded = true;
+              initSegmentObject.value = obj.parsed;
+              return [obj, ...segments];
+            }
+            return [...segments, obj];
+          }, [] as Array<ILoadedSegmentObject<T>>)
+          .mergeMap((segments) => {
+            return initSegmentObject.isDownloaded ?
+              Observable.of(...segments) :
+              Observable.empty() as Observable<never>;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unnecessary cast. ([typescript:S4325](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4325))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=227&issues=AWOTbCHfp7N0pg-JqMCI&open=AWOTbCHfp7N0pg-JqMCI)

[ISSUE_KEY]: AWOTbCHfp7N0pg-JqMCI